wcfsetup/install/files/lib/util/UserAgent.class.php

```diff
@@ -0,0 +1,407 @@
+<?php
+namespace wcf\util;
+
+/**
+ * Represents the user agent.
+ *
+ * @author	Joshua Ruesweg
+ * @copyright	2001-2020 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\Util
+ * @since       5.4
+ */
+class UserAgent {
+	/**
+	 * @var string 
+	 */
+	public $userAgent;
+	
+	/**
+	 * @var string|null
+	 */
+	public $browser;
+	
+	/**
+	 * @var string|null
+	 */
+	public $browserVersion;
+	
+	/**
+	 * @var string|null
+	 */
+	public $os;
```

Comment:
These should be protected.

Reply:
For which reason?

Reply:
You expose getter for these. There should be no need to touch these properties from the outside. The public API should be as small as possible, expanding visibility is easy, the inverse is not.